R/util-CombineSpecs.R

```diff
@@ -0,0 +1,21 @@
+#' Combine Specifications
+#'
+#' Combine a list of specifications into a single specification.
+#'
+#' @param lSpecs A list of specifications.
+#' @return A list representing the combined specification.
+#' @examples
+#' all_wf <- MakeWorkflowList()
+#' all_specs <- CombineSpecs(all_wf)
+#'
+#' @export
+
+CombineSpecs <- function(lSpecs){
+    all_specs <- list()
+    for (spec in lSpecs){
+        for (domain in names(spec)){
+            all_specs[[domain]] <- c(all_specs[[domain]], spec[[domain]])
```

Comment:
When writing tests for this, I thought of a scenario that i want to make sure i understand the intended use. If `col1` is in both domain 1 and domain 2, then should we (1) de-dupe this list, (2) throw a message in case they are named the same but aren't the same, or (3) just keep all duplicates? Currently the third option is how it behaves.

Reply:
I think we just de-dupe the list. 

More generally, I think it's generally fine (even expected) to have duplicate columns across domains/specs. `siteid` and `subjectid` for example, would be expected in multiple domains. Similarly, it would be fine for a column to be included in specs for the same domain in multiple modules For example, {gsm.app} and {gsm.endpoints} would both expect dfAE to have a `subjectid` column.

Is there another use case you have in mind that I'm not considering?

Reply:
I guess there is a bit of nuance we might need around a scenario where the underlying properties of a column are different across 2 different specs. I guess we would want to do the following: 
- if one spec has `required: true` and the other has `required: false` set the combined spec to `required:true`
- if specs have different `type` the we should use the first `type` and throw a warning about the mismatch. 

I think these probably belong in a separate v2.1 issue.

Reply:
yes, i think we should de-dupe. and as you mention, once we have more properties to a column, we would want to make sure they match (in the case of `type`). for `required` it might be slightly more complicated because we wouldn't want to error out when an optional column is missing from a given df even if it is required elsewhere. i'll file a few issues to address these edits.